framework/src/controller/bus.js

```diff
@@ -62,11 +63,21 @@ class Bus extends EventEmitter2 {
 				.catch(error => setImmediate(cb, error));
 		});
 
-		return new Promise((resolve, reject) => {
-			// TODO: wait for all sockets to be created
-			this.rpcSocket.once('bind', resolve);
-			this.rpcSocket.once('error', reject);
-			this.rpcSocket.bind(socketsPath.rpc);
+		let timeout;
+
+		return Promise.race([
```

Comment:
I would try to not have promise declaration inside the race here to make the code more readable.

Suggestion:
```
const milliseconds = 2000;
return Promise.race([
	this._resolveWhenAllSucceed(),
	this._rejectWhenFirstFaill(),
	this._rejectWhenTimeout(milliseconds),
]);
```

We don't need to call `this.cleanup()` inside the timeout because the whole method will be reject and handled in the application level meaning we don't need the `finally` as well.
Note that I slightly changed the function names to better reflect its behaviour.